src/LiveComponent/assets/src/Component/ValueStore.ts

```diff
@@ -50,6 +50,9 @@ export default class {
         }
 
         if (this.props[normalizedName] !== undefined) {
+            if (Array.isArray(this.props[normalizedName])) {
+                return [...this.props[normalizedName]];
+            }
```

Comment:
Instead of this change, I'm thinking we should:

```diff
--- a/src/LiveComponent/assets/src/dom_utils.ts
+++ b/src/LiveComponent/assets/src/dom_utils.ts
@@ -266,24 +266,25 @@ export function getElementAsTagText(element: HTMLElement): string {
 }

 const getMultipleCheckboxValue = function (element: HTMLInputElement, currentValues: Array<string>): Array<string> {
+    const finalValues = [...currentValues]
     const value = inputValue(element);
     const index = currentValues.indexOf(value);

     if (element.checked) {
         // Add value to an array if it's not in it already
         if (index === -1) {
-            currentValues.push(value);
+            finalValues.push(value);
         }

-        return currentValues;
+        return finalValues;
     }

     // Remove value from an array
     if (index > -1) {
-        currentValues.splice(index, 1);
+        finalValues.splice(index, 1);
     }

-    return currentValues;
+    return finalValues;
 };

 const inputValue = function (element: HTMLInputElement): string {
```

Currently (right or wrong), `ValueStore` returns (in the case of arrays/objects) a reference to the *actual* data in the `ValueStore`. This changes that to return a clone. The above change is smaller/safer for the time being. It fixes this exact checkbox case where we were accidentally modifying the value store directly.

I think modifying the `ValueStore` to not return references might, indeed, be a good idea (it would avoid an accident like this). But safer to do it later.